work/load/load_processors.go

```diff
@@ -0,0 +1,175 @@
+package load
+
+import (
+	"context"
+	"fmt"
+	"math/rand/v2"
+	"slices"
+	"time"
+
+	"github.com/tidepool-org/platform/errors"
+	"github.com/tidepool-org/platform/pointer"
+	work "github.com/tidepool-org/platform/work"
+)
+
+const (
+	quantity                     = 1
+	frequency                    = 5 * time.Second
+	processingAvailableTimeDelay = 5 * time.Second
+	processingTimeout            = 5 * 60
+
+	MetadataProcessResult = "processResult"
+)
+
+func domainName(subdomain string) string {
+	return fmt.Sprintf("org.tidepool.work.test.load.%s", subdomain)
+}
+
+// TypeSleepy - will sleep for a random amount of time from 0 - `sleepMaxMillis` and then returns `ResultDelete`
+var TypeSleepy = domainName("sleepy")
+
+// TypeDopey - has to be told what to do or by default returns `ResultDelete`
+var TypeDopey = domainName("dopey")
+
+type loadProcessor struct {
+	pType string
+}
+
+func newLoadProcessor(pType string) (*loadProcessor, error) {
+	if pType != TypeSleepy && pType != TypeDopey {
+		return nil, errors.Newf("type %s is invalid. Must be one of [%s,%s]", pType, TypeSleepy, TypeDopey)
+	}
+	return &loadProcessor{
+		pType: pType,
+	}, nil
+}
+
+func (p *loadProcessor) Type() string {
+	return p.pType
+}
+
+func (p *loadProcessor) Quantity() int {
+	return quantity
+}
+
+func (p *loadProcessor) Frequency() time.Duration {
+	return frequency
+}
+
+func (p *loadProcessor) getProcessResult(wrk *work.Work, result any, errMsg *string) *work.ProcessResult {
+	switch result.(string) {
+	case work.ResultFailing:
+
+		msg := "failing from load test"
+		if errMsg != nil {
+			msg = *errMsg
+		}
+
+		return work.NewProcessResultFailing(work.FailingUpdate{
```

Comment:
You will likely want to add a error and retry time to this. Also, I'd recommend incrementing the retry count each time.